docs/side_quests/working_with_files.md

```diff
@@ -0,0 +1,898 @@
+
+We can accomplish this in a single `map` operation by using Groovy's destructuring assignment. This allows us to unpack the tokenized filename components directly into separate variables, which we can then use to construct our desired tuple structure.
+
+The line `def (sample, replicate, type, readNum) = myFile.simpleName.tokenize('_')` splits the filename at each underscore and assigns each part to a separate variable. We then return these variables along with the file object as a single flat list.
+
+_Before_:
+
+```groovy title="main.nf" linenums="4"
+ch_fastq.map { myFile ->
+    [ myFile.simpleName.tokenize('_'), myFile ]
+}
+```
+
+_After_:
+
+```groovy title="main.nf" linenums="4"
+ch_fastq.map { myFile ->
+    def (sample, replicate, type, readNum) = myFile.simpleName.tokenize('_')
+    [ sample, replicate, type, readNum, myFile ]
+}
+```
+
+```bash
+nextflow run main.nf
+```
+
+```console title="Flattened Metadata Output"
+ N E X T F L O W   ~  version 24.10.4
+
+Launching `main.nf` [extravagant_laplace] DSL2 - revision: 297959a496
+
+[sampleA, rep1, normal, R1, /workspaces/training/side-quests/working_with_files/data/sampleA_rep1_normal_R1_001.fastq.gz]
+[sampleA, rep1, normal, R2, /workspaces/training/side-quests/working_with_files/data/sampleA_rep1_normal_R2_001.fastq.gz]
+```
+
+Success! We've now flattened the metadata into a single list of values.
+
+### 3.4. Simplifying the metadata
+
+The metadata includes a lot of redundant information such as "rep". Let's remove this so replicate is just a number.
+
+We can do this by using `.replace()` on the replicate string to remove the "rep" prefix.
+
+_Before_:
+
+```groovy title="main.nf" linenums="4"
+ch_fastq.map { myFile ->
+    def (sample, replicate, type, readNum) = myFile.simpleName.tokenize('_')
+    [ sample, replicate, type, readNum, myFile ]
+}
+.view()
+```
+
+_After_:
+
+```groovy title="main.nf" linenums="4"
+ch_fastq.map { myFile ->
+    def (sample, replicate, type, readNum) = myFile.simpleName.tokenize('_')
+    [ sample, replicate.replace('rep', ''), type, readNum, myFile ]
+}
+```
+
+```bash
+nextflow run main.nf
+```
+
+```console title="Simplified Metadata Output"
+ N E X T F L O W   ~  version 24.10.4
+
+Launching `main.nf` [reverent_volta] DSL2 - revision: b3aac71fea
+
+[sampleA, 1, normal, R1, /workspaces/training/side-quests/working_with_files/data/sampleA_rep1_normal_R1_001.fastq.gz]
+[sampleA, 1, normal, R2, /workspaces/training/side-quests/working_with_files/data/sampleA_rep1_normal_R2_001.fastq.gz]
+```
+
+Nice! we have removed the "rep" from the replicate string.
+
+### Takeaway
+
+- We can handle filenames in Nextflow with the power of a full programming language
+- We can treat the filenames as strings to extract relevant information
+- Use of Groovy methods like `tokenize()` and `replace()` allows us to manipulate strings in the filename
```

Comment:
above it says tokenize is a nextflow method. I think form the docs here: https://www.nextflow.io/docs/latest/reference/stdlib.html I think we should be consistent here to not cause more confusion between nextflow vs groovy